balboa-agent/src/main/scala/HttpMetricQueue.scala

```diff
@@ -0,0 +1,61 @@
+package com.socrata.balboa.agent
+
+import java.net.URL
+import java.util.Date
+
+import com.socrata.balboa.metrics.{EntityJSON, Metric, MetricJSON}
+import com.socrata.metrics.{IdParts, MetricQueue}
+import com.stackmob.newman.ApacheHttpClient
+import com.stackmob.newman.dsl.POST
+import com.stackmob.newman.response.HttpResponseCode.Ok
+import com.typesafe.scalalogging.slf4j.StrictLogging
+import org.json4s.{DefaultFormats, Extraction, Formats}
+import org.json4s.jackson.JsonMethods.{pretty, render}
+
+import scala.concurrent.duration.Duration
+import scala.concurrent.{Await, ExecutionContext, Future}
+import scala.util.{Failure, Success}
+
+case class HttpMetricQueue(balboaHttpURL: String, timeout: Duration) extends MetricQueue with StrictLogging {
+
+  implicit val httpClient = new ApacheHttpClient
+  implicit val jsonFormats: Formats = DefaultFormats
+  implicit val executionContext = ExecutionContext.global
+
+  /**
+   * Interface for receiving a Metric
+   *
+   * @param entity Entity which this Metric belongs to (ex: a domain).
+   * @param name Name of the Metric to store.
+   * @param value Numeric value of this metric.
+   * @param timestamp Time when this metric was created.
+   * @param recordType Type of metric to add, See [[Metric.RecordType]] for more information.
+   */
+  override def create(entity: IdParts,
+             name: IdParts,
+             value: Long,
+             timestamp: Long = new Date().getTime,
+             recordType: Metric.RecordType = Metric.RecordType.AGGREGATE): Unit = {
+
+    // TODO: lazy load metrics for the same entity and send them over in batches
```

Comment:
If there are a lot of metrics being sent with the same entity, they can be send over in the same POST request rather than each one getting its own HTTP request. It would cut down on overhead, but it's not necessary for functionality.